components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtImpl.java

```diff
@@ -209,6 +209,33 @@ public void activateTenant(String tenantUniqueID) throws TenantMgtException {
                 PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername() + "'");
     }
 
+    @Override
+    public void activateTenant(int tenantId) throws TenantMgtException {
+
+        TenantManager tenantManager = TenantMgtServiceComponent.getTenantManager();
+        String tenantDomain = null;
+        try {
+            Tenant tenant = (Tenant) tenantManager.getTenant(tenantId);
+            if (tenant != null) {
+                tenantManager.activateTenant(tenantId);
+                tenantDomain = tenant.getDomain();
+                // Notify tenant activation to all listeners.
+                TenantMgtUtil.triggerTenantActivation(tenantId);
+                log.info(String.format("Tenant with id: %s and domain: %s activated by %s", tenantId,
+                        tenantDomain, PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername()));
+            } else {
+                throw new TenantManagementClientException(ERROR_CODE_RESOURCE_NOT_FOUND.getCode(),
+                        String.format(ERROR_CODE_RESOURCE_NOT_FOUND.getMessage(), tenantId));
+            }
+        } catch (org.wso2.carbon.user.api.UserStoreException e) {
+            throw new TenantManagementServerException(String.format("Error while retrieving or deactivating the " +
+                    "tenant with id: %s", tenantId), e);
+        } catch (StratosException e) {
+            throw new TenantManagementServerException(String.format("Error in notifying tenant activation of tenant " +
```

Comment:
`tenantDomain` could be null if `activateTenant()` fails.

Reply:
activateTenant() throws a UserStoreException. StratosException is thrown from triggerTenantActivation method. hence tenantDomain will not be null.